packages/cadl-ranch-specs/http/azure/client-generator-core/internal/mockapi.ts

```diff
@@ -1,31 +1,22 @@
-import { passOnSuccess, mockapi, ValidationError, json } from "@azure-tools/cadl-ranch-api";
+import { passOnSuccess, mockapi, ValidationError, json, MockApi } from "@azure-tools/cadl-ranch-api";
 import { ScenarioMockApi } from "@azure-tools/cadl-ranch-api";
 
 export const Scenarios: Record<string, ScenarioMockApi> = {};
 
-Scenarios.Azure_ClientGenerator_Core_Internal_getInternal = passOnSuccess(
-  mockapi.get("/azure/client-generator-core/internal/getInternal", (req) => {
+function createInternalMockApis(route: string): MockApi {
+  const url = `/azure/client-generator-core/internal/${route}`;
+  return mockapi.get(url, (req) => {
     if (!("name" in req.query)) {
       throw new ValidationError("Should submit name query", "any string", undefined);
     }
     return {
       status: 200,
       body: json({ name: req.query["name"] }),
     };
-  }),
-);
+  });
+}
 
-Scenarios.Azure_ClientGenerator_Core_Internal_postInternal = passOnSuccess(
-  mockapi.post("/azure/client-generator-core/internal/postInternal", (req) => {
-    if (req.body["id"] !== 1) {
-      throw new ValidationError("Should submit body id", "1", undefined);
-    }
-    if (!("name" in req.body)) {
-      throw new ValidationError("Should submit body name", "any string", undefined);
-    }
-    return {
-      status: 200,
-      body: json({ id: 1, name: req.body["name"] }),
-    };
-  }),
-);
+Scenarios.Azure_ClientGenerator_Core_Internal_publicOnly = passOnSuccess(createInternalMockApis("public"));
+Scenarios.Azure_ClientGenerator_Core_Internal_internalOnly = passOnSuccess(createInternalMockApis("internal"));
+Scenarios.Azure_ClientGenerator_Core_Internal_Shared_public = passOnSuccess(createInternalMockApis("shared/public"));
```

Comment:
Those 2 should be under the same scenario you can pass a list to `passOnSuccess`